Update local openstack-cleanvm

jgrassler opened the pull request:
This pull request polishes up the scripts/heat template for deploying `openstack-cleanvm` locally.

* Added default repository URLs/branches (default is to build Ocata cloud
  on SLES 12 SP3)
* Adjusted defaults in the Heat template to build an Ocata based cloud

Caveat: I haven't tested this yet, so it may still be subtly broken. I'll be off for two weeks starting next week so won't be able to fix it for a while if it does turn out to be broken. Feel free to play with this/fix it/merge it meanwhile. Failing that I'll test/fix it in two weeks :-)

jdsn commented:
I guess we wait for @jgrassler to test it before merging it, right?

toabctl commented:
> I guess we wait for @jgrassler to test it before merging it, right?

I assume he tested it already. I'm fine with merging it now.

jgrassler commented:
Sorry, I forgot this still needed testing. Let me quickly spin up an instance..

jgrassler commented on jenkins/ci.opensuse.org/heat-templates/deploy-cleanvm.sh:
```diff
@@ -50,7 +54,12 @@ case "$VERSION_ID" in
-zypper --non-interactive install git-core ca-certificates-mozilla
+$zypper install git-core ca-certificates-mozilla
+
+# Make sure these packages are not present. If they present on the system, an
```
This is not exactly elegant. I'd prefer setting allowVendorChange=True for cleanvm in general to circumvent this class of problem entirely. Thoughts?

jgrassler commented:
Testing revealed some issues, please have another look at the card. There were two problems:

The first is rather trivial and easy to fix: in the Heat template's parameters the default release wasn't `Ocata` and the default image wasn't `SLE 12 SP3`. I fixed that part.

The second is a bit hairier. The SLE 12 SP3 image on our local OpenStack cloud comes with a bunch of packages pre-installed that we also have in `Devel:Cloud:Ocata` (I haven't tested it, but the problem should affect Pike as well). When zypper discovers this it stops to ask about a vendor change, thus stalling cleanvm deployment. I hacked up a solution that fixes the situation for these specific packages, but I'd prefer the generic solution I outlined in the inline comment above. Any objections?

jgrassler commented:
Here's the general solution: https://github.com/SUSE-Cloud/automation/pull/2196

I removed the workaround on here for now. Please speak up if you prefer the workaround after all.

jgrassler commented:
Ok, tested the general solution and it's looking good. You can test it yourself as follows:

```
openstack --insecure stack create --wait --parameter key_name=jgrassler --parameter automation_repo=https://github.com/jgrassler/automation.git --parameter automation_branch=cleanvm-allow-vendor-change -t openstack-cleanvm.yaml cleanvm-ocata-jgrassler; openstack --insecure stack output show cleanvm-ocata-jgrassler floating_ip
```

(substitute your own key/stack name)

jgrassler commented:
Ok, I adjusted this to set `allow_vendor_change=1` now (see https://github.com/SUSE-Cloud/automation/pull/2196 ). Also, I updated to a default `cloudsource=openstackpike` since we are focusing on Pike now.
